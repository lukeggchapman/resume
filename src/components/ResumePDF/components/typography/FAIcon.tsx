/**
 * FAIcon is an implementation of Font Awesome Free for use with React-PDF.
 *
 * <FAIcon icon="github" fab /> for brand icons.
 * <FAIcon icon="envelope" far /> for regular icons.
 * <FAIcon icon="envelope" fas /> for solid icons.
 */

import React from 'react'
import { Text, StyleSheet } from '@react-pdf/renderer'

import './fonts/fontawesome'
import fa from './fonts/fontawesome/faUnicode.json'

export const styles = StyleSheet.create({
  icon: {
    fontFamily: 'Font Awesome 5 Free',
  },
  fab: {
    fontWeight: 700,
  },
  far: {
    fontWeight: 400,
  },
  fas: {
    fontWeight: 900,
  },
})

export type IconStyleProps =
  | {
      fab: true
      far?: false
      fas?: false
    }
  | {
      fab?: false
      far: true
      fas?: false
    }
  | {
      fab?: false
      far?: false
      fas: true
    }

export interface FAIconProps {
  icon: keyof typeof fa
}

export default ({ icon, fab, far, fas }: FAIconProps & IconStyleProps) => {
  const faStyle =
    (fab && styles.fab) ||
    (far && styles.far) ||
    (fas && styles.fas) ||
    styles.far

  return <Text style={[styles.icon, faStyle]}>{fa[icon]}</Text>
}
