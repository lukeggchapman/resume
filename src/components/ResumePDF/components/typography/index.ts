export { default } from './typography'
export { default as FAIcon } from './FAIcon'
