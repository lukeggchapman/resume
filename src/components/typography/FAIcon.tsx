/**
 * FAIcon is an implementation of Font Awesome Free for use with React-PDF.
 *
 * <FAIcon icon="github" fab /> for brand icons.
 * <FAIcon icon="envelope" far /> for regular icons.
 * <FAIcon icon="envelope" fas /> for solid icons.
 */

import React from 'react'
import { Text, StyleSheet } from '@react-pdf/renderer'

import './fonts/fontawesome'
import fa from './fonts/fontawesome/faUnicode.json'

export const styles = StyleSheet.create({
  icon: {
    fontFamily: 'Font Awesome 5 Free',
  },
  fab: {
    fontWeight: 700,
  },
  far: {
    fontWeight: 400,
  },
  fas: {
    fontWeight: 900,
  },
})

export type IconStyleProps =
  | {
      fab: true
      far?: false
      fas?: false
    }
  | {
      fab?: false
      far: true
      fas?: false
    }
  | {
      fab?: false
      far?: false
      fas: true
    }

export interface FAIconProps {
  icon: keyof typeof fa
}

const getTrueKey = (obj: IconStyleProps) => {
  const keys = (Object.keys(obj) as unknown) as (keyof IconStyleProps)[]

  return keys.find(key => !!obj[key]) as keyof IconStyleProps
}

export default ({ icon, ...iconStyleProps }: FAIconProps & IconStyleProps) => {
  const faStyle = styles[getTrueKey(iconStyleProps)]

  return <Text style={[styles.icon, faStyle]}>{fa[icon]}</Text>
}
