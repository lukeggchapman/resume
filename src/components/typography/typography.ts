import ReactPDF, { StyleSheet } from '@react-pdf/renderer'

import './fonts/oswald'
import './fonts/notosans'
import './fonts/ptsans'

export default StyleSheet.create({
  body: {
    fontFamily: 'Noto Sans',
    fontSize: 12,
    color: '#333',
  },
  h1: {
    fontFamily: 'Oswald',
    fontWeight: 700,
    fontSize: 24,
  },
  h2: {
    fontFamily: 'Oswald',
    fontSize: 16,
    fontWeight: 900,
    borderBottom: '1 solid #333',
    marginBottom: 5,
  },
  h3: {
    fontFamily: 'Oswald',
    fontSize: 14,
  },
  h4: {
    fontFamily: 'PT Sans',
    fontSize: 8,
    color: '#555',
  },
}) as ReactPDF.Styles
