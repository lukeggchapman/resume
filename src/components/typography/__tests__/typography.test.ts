import typography from '../typography'

describe('PDF typography', () => {
  it('styles are unchanged', () => {
    expect(typography).toMatchSnapshot()
  })
})
