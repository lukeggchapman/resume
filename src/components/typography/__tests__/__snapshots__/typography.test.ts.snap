// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDF typography styles are unchanged 1`] = `
Object {
  "body": Object {
    "color": "#333",
    "fontFamily": "Crimson Text",
    "fontSize": 12,
  },
  "h1": Object {
    "fontFamily": "Oswald",
    "fontSize": 24,
    "fontWeight": 700,
  },
  "h2": Object {
    "borderBottom": "1 solid #333",
    "fontFamily": "Oswald",
    "fontSize": 16,
    "fontWeight": 900,
    "marginBottom": 5,
  },
  "h3": Object {
    "fontFamily": "Oswald",
    "fontSize": 14,
  },
  "h4": Object {
    "color": "#555",
    "fontFamily": "PT Sans",
    "fontSize": 10,
  },
}
`;
